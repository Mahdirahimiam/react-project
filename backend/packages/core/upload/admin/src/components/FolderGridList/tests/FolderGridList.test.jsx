import React from 'react';

import { lightTheme, ThemeProvider } from '@strapi/design-system';
import { render } from '@testing-library/react';
import { IntlProvider } from 'react-intl';
import { MemoryRouter } from 'react-router-dom';

import { FolderGridList } from '../FolderGridList';

const ComponentFixture = () => {
  return (
    <MemoryRouter>
      <IntlProvider locale="en" messages={{}}>
        <ThemeProvider theme={lightTheme}>
          <FolderGridList title="Folder grid list title">Folder grid list children</FolderGridList>
        </ThemeProvider>
      </IntlProvider>
    </MemoryRouter>
  );
};

const setup = (props) => render(<ComponentFixture {...props} />);

describe('FolderGridList', () => {
  it('renders and match snapshots', () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });
});
