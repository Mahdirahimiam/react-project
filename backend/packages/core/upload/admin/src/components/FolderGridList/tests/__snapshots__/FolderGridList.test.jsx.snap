// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FolderGridList renders and match snapshots 1`] = `
.c3 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c1 {
  font-weight: 500;
  font-size: 1rem;
  line-height: 1.25;
  font-weight: 500;
  color: #32324d;
}

.c0 {
  padding-bottom: 8px;
}

.c2 {
  display: grid;
  grid-template-columns: repeat(12,1fr);
  gap: 16px;
}

<div>
  <div
    class=""
  >
    <div
      class="c0"
    >
      <h2
        class="c1"
      >
        Folder grid list title
      </h2>
    </div>
    <div
      class="c2"
    >
      Folder grid list children
    </div>
  </div>
  <div
    class="c3"
  >
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-log"
      role="log"
    />
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-status"
      role="status"
    />
    <p
      aria-live="assertive"
      aria-relevant="all"
      id="live-region-alert"
      role="alert"
    />
  </div>
</div>
`;
